02.calendar/cal.rb

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env ruby
+
+require 'date'
+require 'optparse'
+
+RED = 31
+DEFAULT = 39
+
+year = Date.today.year
+month = Date.today.month
+
+opt = OptionParser.new
+opt.on('-y VAL', String) do |y|
+  year = y.to_i
+end
+
+opt.on('-m VAL') do |m|
+  month = m.to_i
+end
+
+opt.parse!(ARGV)
+
+target_month = Date.new(year, month)
+target_month_last_date = Date.new(year, month, -1)
+days_in_month = [*Array.new(target_month.wday, ''), *1..target_month_last_date.day]
+
+case target_month.strftime('%B').length
+when 3..4
+  format_month = "%15s"
+when 5..6
+  format_month = "%16s"
+when 7..8
+  format_month = "%17s"
+when 9
+  format_month = "%18s"
+end
+
+printf(format_month, "#{target_month.strftime('%B')} #{target_month.year}")
+puts
+printf "Su Mo Tu We Th Fr Sa"
+
+color = DEFAULT
+days_in_month.each_with_index do |day, i|
+  format_day = "%8s"
+  if i % 7 == 0
+    puts
+    format_day = "%7s"
+  end
+
+  if !(day.to_s).empty? && Date.new(target_month.year, target_month.month, day) == (Date.today)
+    color = RED
+    printf(format_day, "\e[#{color}m#{day}")
```

Comment:
```suggestion
    printf(format_day, "\e[#{color}m#{day}\e[0m")
```

にすれば`color = DEFAULT`は不要になるかも？

Reply:
`\e[0m`を末尾に加えてもカラーが元に戻りませんでした、、
![スクリーンショット 2023-06-27 232951](https://github.com/TakahiroIzuka/ruby-practices/assets/67720987/2e3bbd52-c637-48ff-b04f-2c6d1b5d3b48)

Reply:
手元で動かしてみたところ、ちゃんと動いているように見えますね 🤔 

<img width="241" alt="Screenshot 2023-07-03 at 8 52 19" src="https://github.com/TakahiroIzuka/ruby-practices/assets/1148320/3ab871de-b762-4e9c-ac8f-5067cd14876b">